core/src/components/datetime/utils/warn.ts

```diff
@@ -0,0 +1,52 @@
+import { printIonWarning } from '@utils/logging';
+
+import type { DatetimePresentation, FormatOptions } from '../datetime-interface';
+
+/**
+ * If a time zone is provided in the format options, the rendered text could
+ * differ from what was selected in the Datetime, which could cause
+ * confusion.
+ */
+export const warnIfTimeZoneProvided = (formatOptions?: FormatOptions) => {
+  if (
+    formatOptions?.date?.timeZone ||
+    formatOptions?.date?.timeZoneName ||
+    formatOptions?.time?.timeZone ||
+    formatOptions?.time?.timeZoneName
+  ) {
+    printIonWarning('Datetime: "timeZone" and "timeZoneName" are not supported in "formatOptions".');
+  }
+};
+
+export const checkForPresentationFormatMismatch = (
+  presentation: DatetimePresentation,
+  formatOptions?: FormatOptions
+) => {
+  // formatOptions is not required
+  if (!formatOptions) return;
+
+  // If formatOptions is provided, the date and/or time objects are required, depending on the presentation
+  switch (presentation) {
+    case 'date':
+    case 'month-year':
+    case 'month':
+    case 'year':
+      if (formatOptions.date === undefined) {
+        printIonWarning(`Datetime: The '${presentation}' presentation requires a date object in formatOptions.`);
+      }
+      break;
+    case 'time':
```

Comment:
mainly for my understanding: This is still needed for the upcoming `ion-datetime-button` integration right? For `presentation="time"` there's nothing in the datetime UI that would need `formatOptions`: 
![image](https://github.com/ionic-team/ionic-framework/assets/2721089/f75c851c-c667-48bc-a885-090ee990d88f)


However, this would be needed for the time display in datetime button. Am I understanding correctly?

Reply:
That's correct!